app/Espo/Core/Templates/Repositories/Hierarchy.php

```diff
@@ -398,6 +398,32 @@ public function getParentRecord(string $id): array
         return empty($record) ? [] : $record;
     }
 
+    protected function entityHasArchive(Entity $entity): bool
+    {
+        return !empty($this->getMetadata()->get(['scopes', $entity->getEntityType(), 'hasArchive']));
+    }
+
+
+    protected function validateIsArchived(Entity $entity): void
+    {
+        $fieldName = 'isArchived';
+        if ($entity->isAttributeChanged($fieldName) && $entity->get($fieldName) == true) {
+            // search all childs
+            $hasNonArchivedChildren = false;
+            foreach ($entity->get('children') as $child) {
+                if ($child->get('is_archived') == false) {
```

Comment:
Why ```is_archived``` in underscore? Should be isArchived

Reply:
Fixed